src/main/java/soot/AbstractSootFieldRef.java

```diff
@@ -34,15 +34,17 @@
  * actually exist; the actual target of the reference is determined according to the resolution procedure in the Java Virtual
  * Machine Specification, 2nd ed, section 5.4.3.2.
  */
-
 public class AbstractSootFieldRef implements SootFieldRef {
   private static final Logger logger = LoggerFactory.getLogger(AbstractSootFieldRef.class);
 
+  private final SootClass declaringClass;
+  private final String name;
+  private final Type type;
+  private final boolean isStatic;
+
+  private SootField resolveCache = null;
```

Comment:
as there can exist multiple instances of *FieldRef refering to the same SootField - maybe we can use a global cache for it?

Reply:
That's true. I did see a performance improvement with this local caching and had not even considered global caching. I guess the same would apply to the SootMethodRefImpl caching in PR #1520.